### PR TITLE
Reinstate refresh_token tests.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: cloudSupport/src/test/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthModeSpec.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    val privateKey = keyPair.getPrivate
+    val byteEncoded = privateKey.getEncoded
+    val base64Encoded = Base64.getEncoder.encodeToString(byteEncoded)
+    s"""|-----BEGIN PRIVATE KEY-----
+        |$base64Encoded
+        |-----END PRIVATE KEY-----
+        |""".stripMargin
+  }
+
+  // Hide me from git secrets false positives
```
Comment (mcovarr) [inline cloudSupport/src/test/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthModeSpec.scala]: :rofl:

File: common/src/main/scala/common/validation/Validation.scala [hunk 2]
```diff
@@ -40,10 +40,8 @@ object Validation {
   }
 
   implicit class ValidationTry[A](val e: ErrorOr[A]) extends AnyVal {
-    def toTry: Try[A] = e match {
-      case Valid(options) => Success(options)
-      case Invalid(err) => Failure(AggregatedMessageException("Error(s)", err.toList))
-    }
+    def toTry: Try[A] = toTry("Error(s)")
```
Comment (mcovarr) [inline common/src/main/scala/common/validation/Validation.scala]: so happy there isn't a separate PR in a separate repo for this 😂

File: common/src/test/scala/common/validation/ValidationSpec.scala [hunk 3]
```diff
@@ -78,4 +79,14 @@ class ValidationSpec extends FlatSpec with Matchers {
     failure.toChecked shouldBe Left(NonEmptyList.of(":("))
   }
 
+  it should "convert an ErrorOr to a Try" in {
+    val valid = "yeah".valid
+    val invalid = ":(".invalidNel
```
Comment (mcovarr) [inline common/src/test/scala/common/validation/ValidationSpec.scala]: a unicode frowny might be more clear 😄

File: cloudSupport/src/main/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthMode.scala [hunk 4]
```diff
@@ -81,14 +71,26 @@ sealed trait GoogleAuthMode {
 
   def requiresAuthFile: Boolean = false
 
+  // changing this is for testing only. all the inheritance is otherwise locked down.
```
Comment (geoffjentry) [inline cloudSupport/src/main/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthMode.scala]: I think I understand what this sentence is saying but I'm not sure, which makes me think it should be rephrased somehow

File: cloudSupport/src/main/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthMode.scala [hunk 5]
```diff
@@ -81,14 +71,26 @@ sealed trait GoogleAuthMode {
...
   protected def validateCredential(credential: Credentials) = {
-    Try(credential.refresh()) match {
+    Try(credentialValidation(credential)) match {
       case Failure(ex) => throw new RuntimeException(s"Google credentials are invalid: ${ex.getMessage}", ex)
       case Success(_) => credential
     }
   }
 }
 
+case object MockAuthMode extends GoogleAuthMode {
```
Comment (geoffjentry) [inline cloudSupport/src/main/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthMode.scala]: Is this used only in test code? if so would it make more sense to live over in test-land?

Comment (kshakir) [reply cloudSupport/src/main/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthMode.scala]: That would be nice, but I don't think that works with `sealed trait`s. In this PR I just moved the class from an inner-class to a top-level-class.

**Comment by gemlam3** [comment]
@geoffjentry can you re-review this and approve if it looks good?

**Pull request merged**